misc/python/materialize/parallel_workload/parallel_workload.py

```diff
@@ -0,0 +1,139 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+import argparse
+import datetime
+import os
+import random
+import sys
+import threading
+import time
+from typing import Optional
+
+import pg8000
+
+from materialize.parallel_workload.action import ddl_actions, dml_actions
+from materialize.parallel_workload.database import Database
+from materialize.parallel_workload.worker import Worker
+
+SEED_RANGE = 1_000_000
+REPORT_TIME = 10
+
+
+def run(
+    host: str,
+    port: int,
+    seed: str,
+    runtime: int,
+    complexity: str,
+    num_threads: Optional[int],
+) -> None:
+    num_threads = num_threads or os.cpu_count() or 10
+    print(f"Seed: {seed}")
+    random.seed(seed)
+
+    end_time = (
+        datetime.datetime.now() + datetime.timedelta(seconds=runtime)
+    ).timestamp()
+
+    rng = random.Random(random.randrange(SEED_RANGE))
+    database = Database(rng, seed)
+    conn = pg8000.connect(host=host, port=port, user="materialize")
+    conn.autocommit = True
+    with conn.cursor() as cur:
+        database.create(cur)
+    conn.close()
+
+    conn = pg8000.connect(
+        host=host, port=port, user="materialize", database=str(database)
+    )
+    conn.autocommit = True
+    with conn.cursor() as cur:
+        database.create_relations(cur)
+
+    workers = []
+    threads = []
+    for i in range(num_threads):
+        worker_rng = random.Random(rng.randrange(SEED_RANGE))
+        chances = [80, 20] if complexity == "ddl" else [100, 0]
+        actions = [
+            (action[0](worker_rng, database), action[1])
```

Comment:
There are quite a bit of subscript-ing going on around `action`, with `[0]` meaning the action and `[1]` meaning its probability. You may wish to define named constants for 0 or 1 or use a lightweight class so that those two items of the array can be given names.